plugins/inputs/turbostat/plugin.go

```diff
@@ -0,0 +1,87 @@
+package turbostat
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Turbostat struct {
+	Command []string `toml:"command"`
+
+	Log telegraf.Logger
+
+	cancel context.CancelFunc
+}
+
+func (*Turbostat) Gather(telegraf.Accumulator) error {
+	return nil
+}
+
+func (*Turbostat) SampleConfig() string {
+	return ""
+}
+
+// Starts a child process and goroutines to process stdout/stderr.
+func (t *Turbostat) Start(a telegraf.Accumulator) error {
+	if len(t.Command) < 1 {
+		return errors.New("no command provided")
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	t.cancel = cancel
+	cmd := exec.CommandContext(ctx, t.Command[0], t.Command[1:]...)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	err = cmd.Start()
+	if err != nil {
+		return err
+	}
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		err = processStdout(stdout, a)
+		if err != nil {
+			t.Log.Errorf("error processing stdout: %s", err)
+			cancel()
+		}
+	}()
+	go func() {
+		defer wg.Done()
+		err = processStderr(stderr, t.Log)
+		if err != nil {
+			t.Log.Errorf("error processing stderr: %s", err)
+			cancel()
+		}
+	}()
+	go func() {
+		wg.Wait()
+		err := cmd.Wait()
+		if err != nil {
+			t.Log.Errorf("child process stopped: %s", err)
+		}
+		t.Log.Error("metrics emission stopped")
+	}()
+	return nil
```

Comment:
Currently, if for some (external?) reason something happens with turbostat output or parsing of that output, turbostat process is terminated, and the plugin stops emitting metrics. 
Would you consider adding functionality to restart turbostat in that case?

Reply:
I implemented a restart with incremental backoff, which resets if the child process lives long enough. This limits the number of times we attempt to start Turbostat in case of early fatal errors (e.g.: misconfiguration, not running with sudo, etc.). At the same time it restart Turbostat reasonably fast in case there is an odd error. I tested killing the process and it worked as expected. I pull a new dependency, but it was already an indirect dependency, and it does not have further dependencies itself.